_assets/js/laws-regs-parser.js

```diff
@@ -158,3 +155,61 @@ function shareLink(e, divType) {
         shareIcon.style.display = 'block';
     }, 2000);
 }
+
+function addCopyEventListeners(btn, divType) {
+    btn.addEventListener('click', (e) => { copyText(e, divType) });
+    btn.addEventListener('mouseover', (e) => { highlightText(e, divType) });
+    btn.addEventListener('mouseleave', (e) => { unHighlightText(e, divType) });
+}
+
+function addPrintEventListeners(btn, divType) {
+    btn.addEventListener('click', (e) => { printText(e, divType) });
+    btn.addEventListener('mouseover', (e) => { highlightText(e, divType) });
+    btn.addEventListener('mouseleave', (e) => { unHighlightText(e, divType) });
+}
+
+function search() {
+    const searchBox = document.getElementById('searchbox');
+    const searchQuery = searchBox.value.toLowerCase();
+    if (!searchQuery.length){
+        clearSearch();
+        return;
+    };
+    const sections = document.querySelectorAll('.section')
+    for (let i = 0; i < sections.length; i++) {
+        removeHighlights(sections[i]);
+        if(sections[i].innerText.toLowerCase()
+            .includes(searchQuery)) {
+            sections[i].classList.remove("hidden");
+            sections[i].classList.add("searched");
+            highlightTerm(searchQuery, sections[i]);
+        } else {
+            sections[i].classList.add("hidden");
+            sections[i].classList.remove("searched");
+        }
+    }
+}
+
+function highlightTerm(text, section) {
+    const innerHTML = section.innerHTML.replaceAll(new RegExp('(' + text + ')',"ig"), `<span class='search-term'>$1</span>`);
+    section.innerHTML = innerHTML;
+    const shareBtn = section.querySelector('.Share-btn');
+    shareBtn.addEventListener('click', (e) => { shareLink(e, '.section') });
+    const copyBtn = section.querySelector('.Copy-btn');
+    addCopyEventListeners(copyBtn, '.section');
+    const printBtn = section.querySelector('.Print-btn');
+    addPrintEventListeners(printBtn, '.section');
+  }
+
+function removeHighlights(section) {
+    const innerHTML = section.innerHTML.replaceAll('<span class="search-term">', '').replaceAll("</span>", '');
```

Comment:
I'd be worried about doing a replaceAll on the innerHTML, as I don't think there's standardization as to how browsers handle innerHTML spacing? Plus, this isn't extensible if more spans are added to the section html (because of the replacement of `</span>`

an alternative might be to interact with the DOM instead:

`section.querySelectorAll('span.search-term').forEach(term => term.remove());`

Reply:
Experimented with this but it removes the entire element (with the text we need to keep). I then tried to replace the element with the inner text but that also wasn't working so left this as is for now.